test/sentry/opentelemetry/span_processor_test.exs

```diff
@@ -0,0 +1,123 @@
+defmodule Sentry.Opentelemetry.SpanProcessorTest do
+  use Sentry.Case, async: true
+
+  import Sentry.TestHelpers
+
+  alias Sentry.OpenTelemetry.SpanStorage
+
+  defmodule TestEndpoint do
+    require OpenTelemetry.Tracer, as: Tracer
+
+    def instrumented_function do
+      Tracer.with_span "instrumented_function" do
+        :timer.sleep(100)
+
+        child_instrumented_function("one")
+        child_instrumented_function("two")
+      end
+    end
+
+    def child_instrumented_function(name) do
+      Tracer.with_span "child_instrumented_function_#{name}" do
+        :timer.sleep(140)
+      end
+    end
+  end
+
+  @tag span_storage: true
+  test "sends captured root spans as transactions" do
+    put_test_config(environment_name: "test", traces_sample_rate: 1.0)
```

Comment:
The config we use is global, so you cannot use `async: true` for the test. Do we do that in any other test?